crates/proto/src/rr/domain/name.rs

```diff
@@ -1928,4 +1957,118 @@ mod tests {
         assert!(iter.next().is_none());
         assert_eq!(iter.size_hint().0, 0);
     }
+
+    #[test]
+    fn test_label_randomization() {
+        let mut name = Name::root();
+        name.randomize_label_case();
+        assert!(name.eq_case(&Name::root()));
+
+        for qname in [
+            "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa.",
+            "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA.",
+            "abcdefghijklmnopqrstuvwxyz0123456789.",
+            "ABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890.",
+            "www01.example-site.com",
+            "1234567890.e-1204089_043820-5.com.",
+        ] {
+            let mut name = Name::from_ascii(qname).unwrap();
+            let name2 = name.clone();
+            name.randomize_label_case();
+            assert_eq!(name, name2);
+            println!("{name2} == {name}: {}", name == name2);
+        }
+
+        // 50k iterations gets us very close to a 50/50 upper/lower distribution in testing without a
+        // super long test runtime.
+        let iterations = 50_000;
+
+        // This is a max length label (255 bytes) with the maximum number of possible flippable bytes
+        // (nominal label length 63, except the last, with all label characters alpha)
+        let test_str = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyzabcdefghijk.lmnopqrstuvwxyzabcdefghjijklmnopqrstuvwxyzabcdefghijklmnopqrstu.vwxyzABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZABCDEF.GHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOPQRSTUVWXYZABCDEFGHIJKLMNOP";
+        let mut name = Name::from_ascii(test_str).unwrap();
+        let name2 = name.clone();
+
+        let len = name.label_data.len();
+        let mut cap_table: [u32; 255] = [0; 255];
+        let mut lower_table: [u32; 255] = [0; 255];
+        let mut mean_table: [f64; 255] = [0.0; 255];
+
+        for _ in 0..iterations {
+            name.randomize_label_case();
+            assert_eq!(name, name2);
+
+            for (j, cbyte) in name.label_data.iter().enumerate() {
+                if *cbyte >= 0x61 && *cbyte <= 0x7a {
```

Comment:
`is_ascii_lowercase()` and `is_ascii_uppercase()`? (Also maybe prefix `cbyte` with `&` in the loop header?)